closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -56,15 +121,159 @@ static void handleErrors(void) {
 
     printf("An error occurred\n");
 
-    while(errCode = ERR_get_error())
+    while(errCode = (*OSSL_get_error)())
     {
-        char *err = ERR_error_string(errCode, NULL);
+        char *err = (*OSSL_error_string)(errCode, NULL);
         printf("Generating error message\n" );
         printf("%s\n", err);
     }
     abort();
 }
 
+/*
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    loadCrypto
+ * Signature: ()V
+ */
+JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+  (JNIEnv *env, jclass thisObj){
+
+    void *handle;
+    char *error;
+    typedef const char* OSSL_version_t(int);
+
+     // Determine the version of OpenSSL.
+    OSSL_version_t* OSSL_version;
+    const char * openssl_version;
+    int ossl_ver;
+
+
+    // Load OpenSSL Crypto library
+    handle = load_crypto_library();
+    if (handle == NULL) {
+        fprintf(stderr, "FAILED TO LOAD OPENSSL CRYPTO LIBRARY\n");
```

Comment:
I don't think the code should be printing to the console when it fails to load the crypto library, it should silently fail.